barman/backup.py

```diff
@@ -558,6 +559,11 @@ def delete_backup(self, backup, skip_wal_cleanup_if_standalone=True):
                 parent_backup.children_backup_ids = None
             parent_backup.save()
 
+        # Update next rsync backup size information
+        # This assumes that rsync can create hard-links to postgres backups
```

Comment:
```suggestion
        # rsync backups can have deduplication at filesystem level by using
        # "reuse_backup = link". The deduplication size is calculated at the
        # time the backup is taken. If we remove a backup, it may be the case
        # that the next backup in the catalog is a rsync backup which was taken
        # with the "link" option. With that possibility in mind, we re-calculate the
        # deduplicated size of the next rsync backup because the removal of the
        # previous backup can impact on that number.
        # Note: we have no straight forward way of identifying if the next rsync
        # backup in the catalog was taken with "link" or not because
        # "reuse_backup" value is not stored in the "backup.info" file. In any
        # case, the "re-calculation" can still be performed, and the only
        # drawback is that we will waste some (small) amount of CPU/disk
        # usage.
```

@bleidens @gcalacoci I think this provides a more clear idea here.
What do you think?

Reply:
When writing comments to the code, let's first make sure they are needed, and if needed, that they provide enough information for us when reviewing the code in the future.